subsys/bluetooth/mesh/adv.c

```diff
@@ -157,25 +251,52 @@ struct net_buf *bt_mesh_adv_buf_get(k_timeout_t timeout)
 	return process_events(events, ARRAY_SIZE(events));
 }
 
-static struct net_buf *bt_mesh_adv_buf_relay_get(k_timeout_t timeout)
-{
-	return net_buf_get(&bt_mesh_relay_queue, timeout);
-}
-
 struct net_buf *bt_mesh_adv_buf_get_by_tag(uint8_t tag, k_timeout_t timeout)
 {
 	if (tag & BT_MESH_LOCAL_ADV) {
 		return bt_mesh_adv_buf_get(timeout);
 	} else if (tag & BT_MESH_RELAY_ADV) {
-		return bt_mesh_adv_buf_relay_get(timeout);
+		return bt_mesh_adv_buf_relay_get();
 	} else {
 		return NULL;
 	}
 }
+
+static void bt_mesh_relay_send(struct net_buf *buf)
+{
+	uint8_t prio = BT_MESH_ADV(buf)->prio;
+	sys_sfnode_t *curr, *prev = NULL;
+
+	if (!IS_ENABLED(CONFIG_BT_MESH_RELAY_PRIORITY) || !prio) {
+		k_queue_append(&bt_mesh_relay_queue, net_buf_ref(buf));
+		bt_mesh_adv_buf_relay_ready();
+		return;
+	}
+
+	k_mutex_lock(&bt_mesh_relay_mutex, K_FOREVER);
+
+	SYS_SFLIST_FOR_EACH_NODE(&bt_mesh_relay_queue.data_q, curr) {
+		struct net_buf *buf_curr = CONTAINER_OF(curr, struct net_buf, node);
+
+		if (BT_MESH_ADV(buf_curr)->prio < prio) {
+			break;
+		}
+
+		prev = curr;
+	}
+
+	/* The messages with the highest priority are always placed at the head, and
+	 * the messages with the same priority are arranged in chronological order.
+	 */
+	k_queue_insert(&bt_mesh_relay_queue, prev, net_buf_ref(buf));
```

Comment:
I think this should be part of `net_buf` API to guarantee that nothing brakes if for some reason `net_buf` implementation changes. Same for `k_queue_get`.

Reply:
I'm not sure if this breaks net_ Buf design, net_ Buf inherits FIFO, which is obviously a first in, first out design.

Reply:
To be sure that it won't break, I think this needs to be part of `net_buf` API.

Reply:
I thought about it again, your purpose is to maintain the insert operation in the net_buf API, but what needs to be considered is that the current net_buf is based on FIFO, I don't want to destroy this structure, my final solution is Abandon net_buf and use memslab, so that node nodes can be managed by us, and part of the ram resource overhead can also be solved.

Reply:
Well, done now.